server/src/test/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapperTests.java

```diff
@@ -88,14 +102,14 @@ public void testDefaults() throws Exception {
 
         List<IndexableField> fields = doc1.rootDoc().getFields("field");
         assertEquals(2, fields.size());
-        assertThat(fields.get(0), Matchers.instanceOf(FeatureField.class));
```

Comment:
Lets add a test that checks that the current lucene version is < 10.2.0, this way we are reminded to remove that XFeatureField eventually once Lucene gets upgraded XFeatureField

Reply:
Added in https://github.com/elastic/elasticsearch/pull/117917/commits/ef0f71064d1d292132351a0567a2c70f647afecd, thanks